src/ucp/core/ucp_worker.c

```diff
@@ -1169,6 +1169,63 @@ void ucp_worker_iface_cleanup(ucp_worker_iface_t *wiface)
     ucp_worker_uct_iface_close(wiface);
 }
 
+static void ucp_worker_close_cms(ucp_worker_h worker)
+{
+    const uint64_t cm_cmpts_bitmap = worker->context->config.cm_cmpts_bitmap;
+    const ucp_rsc_index_t num_cms  = ucs_popcount(cm_cmpts_bitmap);
+    ucp_rsc_index_t i;
+
+    for (i = 0; (i < num_cms) && (worker->cms[i] != NULL); ++i) {
```

Comment:
No, this loop walks thru cm objects, not components. Check for NULL is needed to handle the case if cm was not opened by some reason. Actually it can be replaced with assertion according to current initialization.